### DISCUSSION TITLE
ask where can I read the Dockerfile of official EJBCA docker image

### DISCUSSION THREAD

**Discussion by mulyadisantosa** [Keyfactor/ejbca-ce#318] [Q&A]
Dear all

As the subject says, coukd anyone help to point me where I can get the Dockerfile of official EJBCA docker image (https://hub.docker.com/r/keyfactor/ejbca-ce)?

I need it to study and explore the possibility of doing certain security hardening in the container image itself.

Thank you in advance

regards,

Mulyadi

**Comment by primetomas** [comment]
There is currently not a simple Dockerfile to build the docker image. The container is built as part of a build pipe-line. You have to use other mechanisms to modify an existing container.

**Reply by mulyadisantosa** [reply]
Thank you @primetomas for your hint. 

After further check, I read in the ejbca 7.x branch, such as :
https://github.com/Keyfactor/ejbca-ce/blob/EJBCA_7_6_0_RC1_20210609/jenkins-files/EE_COS7_OpenJDK8_WF14_NOHSM_MariaDB/, there are Dockerfile, Jenkins file etc. Is it the same files EJBCA team use in the build pipe line to produce the official EJBCA docker image?

**Reply by primetomas** [reply]
No, that's not it. Those were used for containers to run system tests in Jenkins, not for production containers. You're free to use those any way you like of course, and build upon it further yourself.
